crates/json/src/schema/formats.rs

```diff
@@ -0,0 +1,88 @@
+    static ref ISO_8601_ONLY_WEEKS_RE: Regex = Regex::new(r"^[0-9P|W]*$").expect("Is a valid regex");
+    static ref ISO_8601_NO_WEEKS_RE: Regex = Regex::new(r"^[^W]*$").expect("Is a valid regex");
+}
+
+pub fn validate_format(format: &String, val: &str) -> bool {
+    match format.as_str() {
+        "date" => {
+            time::Date::parse(
+                val,
+                &time::macros::format_description!("[year]-[month]-[day]"),
+            ).is_ok() &&
+            // Padding with zeroes is ignored by the underlying parser. The most efficient
+            // way to check it will be to use a custom parser that won't ignore zeroes,
+            // but this regex will do the trick and costs ~20% extra time in this validator.
+            DATE_RE
+            .is_match(val).unwrap_or(false)
+        }
+        "date-time" => {
+            time::OffsetDateTime::parse(val, &time::format_description::well_known::Rfc3339).is_ok()
+        }
+        "time" => time::Time::parse(
+            val,
+            &time::macros::format_description!("[hour]:[minute]:[second].[subsecond]Z"),
+        )
+        .is_ok(),
+        "email" => parse_email_address(val).is_ok(),
+        "hostname" => parse_domain_name(val).is_ok(),
+        // The rules/test cases for these are absolutely bonkers
+        // If we end up needing this let's revisit (jshearer)
+        "idn-hostname" | "idn-email" => false,
+        "ipv4" => {
+            if val.starts_with('0') {
+                return false;
+            }
+            match IpAddr::from_str(val) {
+                Ok(i) => i.is_ipv4(),
+                Err(_) => false,
+            }
+        }
+        "ipv6" => match IpAddr::from_str(val) {
+            Ok(i) => i.is_ipv6(),
+            Err(_) => false,
+        },
+        // uuid crate supports non-hyphenated inputs, jsonschema does not
+        "uuid" if val.len() == 36 => Uuid::parse_str(val).is_ok(),
+
+        "duration" => match ISO_8601_DURATION_RE.is_match(val) {
+            Ok(true) => match val.contains("W") {
+                // If we parse as weeks, ensure that ONLY weeks are provided
+                true => ISO_8601_ONLY_WEEKS_RE.is_match(val).unwrap_or(false),
+                // Otherwise, ensure that NO weeks are provided
+                false => ISO_8601_NO_WEEKS_RE.is_match(val).unwrap_or(false),
+            },
+            _ => false,
+        },
+        "iri" => iri_string::validate::iri::<IriSpec>(val).is_ok(),
+        "uri" => iri_string::validate::iri::<UriSpec>(val).is_ok(),
+        "uri-reference" => iri_string::validate::iri_reference::<UriSpec>(val).is_ok(),
+        "iri-reference" => iri_string::validate::iri_reference::<IriSpec>(val).is_ok(),
+        "uri-template" => URI_TEMPLATE_RE.is_match(val).unwrap_or(false),
+        "json-pointer" if !val.get(0..1).map_or(false, |first| first.eq("#")) => {
```

Comment:
nit: `json-pointer` could be a regex pretty easily, removing the dependency. [We do this already in the models crate](https://github.com/estuary/flow/blob/master/crates/models/src/references.rs#L35).

FYI we also have a json-pointer implementation [here in the docs crate](https://github.com/estuary/flow/blob/master/crates/doc/src/ptr.rs). Arguably it should be in the `json` create but :shrug:

Reply:
I tried referencing the `models` crate from `json`, but that created a circular dependency, so I just copied the json pointer regex. How smelly does this feel to you?